mdata/aggmetric_test.go

```diff
@@ -69,10 +73,72 @@ func (c *Checker) Verify(primary bool, from, to, first, last uint32) {
 	cluster.ThisNode.SetPrimary(currentClusterStatus)
 }
 
+func TestMetricPersistBeingPrimary(t *testing.T) {
+	testMetricPersistOptionalPrimary(t, true)
+}
+
+func TestMetricPersistNotBeingPrimary(t *testing.T) {
+	testMetricPersistOptionalPrimary(t, false)
+}
+
+func testMetricPersistOptionalPrimary(t *testing.T, primary bool) {
+	// always reset the counter when entering and leaving the test
+	dnstore.Reset()
+	defer dnstore.Reset()
+
+	cluster.Init("default", "test", time.Now())
+	cluster.ThisNode.SetPrimary(primary)
+
+	callCount := uint32(0)
+	calledCb := make(chan bool)
+	var countingCacheCb cache.CacheCb = func(string, uint32, *chunk.IterGen) {
+		calledCb <- true
+	}
+
+	numChunks, chunkAddCount, chunkSpan := uint32(5), uint32(10), uint32(300)
+	agg := NewAggMetric(dnstore, countingCacheCb, "foo", chunkSpan, numChunks, 1, []AggSetting{}...)
+
+	ts := uint32(1000)
+	for i := uint32(0); i < chunkAddCount; i++ {
+		agg.Add(ts, 1)
+		ts = ts + chunkSpan
+	}
+
+	timeout := make(chan bool, 1)
+	oneSecTimeout := func() {
+		time.Sleep(1 * time.Second)
+		timeout <- true
+	}
+
+	for i := uint32(0); i < chunkAddCount-1; i++ {
```

Comment:
how about we create a mockcache, similar to devnullstore, maybe devnullcache, it could take care of all the counting and the timeouting.